From ee4d0998149b8f270c29b956102bc73064c540da Mon Sep 17 00:00:00 2001
From: Igor Derzhavets <igor.derzhavets@compulab.co.il>
Date: Sun, 28 Jul 2019 19:26:36 +0300
Subject: [PATCH] [CL-SOM-iMX8X] Add build targets for CL-SOM-iMX8X

	make SOC=iMX8QX cl_som_imx8x
	make SOC=iMX8QX cl_som_imx8x_mmc
	make SOC=iMX8QX cl_som_imx8x_flexspi

Signed-off-by: Igor Derzhavets <igor.derzhavets@compulab.co.il>
---
 iMX8QX/soc.mak | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/iMX8QX/soc.mak b/iMX8QX/soc.mak
index 3c8e7b1..932e542 100755
--- a/iMX8QX/soc.mak
+++ b/iMX8QX/soc.mak
@@ -113,6 +113,27 @@ flash_linux_m4_xip: $(MKIMG) mx8qx-ahab-container.img scfw_tcm.bin u-boot-atf-co
                    dd if=u-boot-atf-container.img of=flash.bin bs=1K seek=$$pad_cnt; \
 	./$(QSPI_PACKER) $(QSPI_HEADER)
 
+cl_som_imx8x_mmc: $(MKIMG) mx8qx-ahab-container.img scfw_tcm.bin u-boot-atf-container.img u-boot-spl.bin
+	./$(MKIMG) -soc QX -rev B0 -dcd skip -append mx8qx-ahab-container.img -c -scfw scfw_tcm.bin -ap u-boot-spl.bin a35 0x00100000 -out flash.bin
+	cp flash.bin boot-spl-container.img
+	@flashbin_size=`wc -c flash.bin | awk '{print $$1}'`; \
+		   pad_cnt=$$(((flashbin_size + 0x400 - 1) / 0x400)); \
+		   echo "append u-boot-atf-container.img at $$pad_cnt KB"; \
+		   dd if=u-boot-atf-container.img of=flash.bin bs=1K seek=$$pad_cnt; \
+	cp flash.bin cl_som_imx8x_mmc.bin
+
+cl_som_imx8x_flexspi: $(MKIMG) mx8qx-ahab-container.img scfw_tcm.bin u-boot-spl.bin u-boot-atf-container.img
+	./$(MKIMG) -soc QX -rev B0 -dcd skip -dev flexspi -append mx8qx-ahab-container.img -c -scfw scfw_tcm.bin -ap u-boot-spl.bin a35 0x00100000 -out flash.bin
+	cp flash.bin boot-spl-container.img
+	@flashbin_size=`wc -c flash.bin | awk '{print $$1}'`; \
+                   pad_cnt=$$(((flashbin_size + 0x400 - 1) / 0x400)); \
+                   echo "append u-boot-atf-container.img at $$pad_cnt KB"; \
+                   dd if=u-boot-atf-container.img of=flash.bin bs=1K seek=$$pad_cnt;
+	./$(QSPI_PACKER) $(QSPI_HEADER)
+	cp flash.bin cl_som_imx8x_fspi.bin
+
+cl_som_imx8x: cl_som_imx8x_mmc cl_som_imx8x_flexspi
+
 flash_regression_linux_m4: $(MKIMG) mx8qx-ahab-container.img scfw_tcm.bin u-boot-atf.bin m4_image.bin
 	./$(MKIMG) -soc QX -rev B0 -append mx8qx-ahab-container.img -c -flags 0x00200000 -scfw scfw_tcm.bin -ap u-boot-atf.bin a35 0x80000000 -p3 -m4 m4_image.bin 0 0x34FE0000 -out flash.bin
 
-- 
2.11.0

